src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/scope_map.py

```diff
@@ -0,0 +1,159 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.util import CLIError
+from ._utils import get_resource_group_name_by_registry_name
+
+
+def _validate_and_generate_actions_for_repositories(allow_or_deny_respository):
+    actions = []
+
+    for rule in allow_or_deny_respository:
+        splitted = rule.split(';', 1)
+        if len(splitted) != 2:
+            return False, rule
+        repository, repository_actions = splitted[0], splitted[1].split(',')
+        for action in repository_actions:
+            actions.append("repositories/" + repository + "/" + action)
+
+    return True, actions
+
+
+def acr_scope_map_create(cmd,
+                         client,
+                         registry_name,
+                         scope_map_name,
+                         add_repository,
+                         resource_group_name=None,
+                         description=None):
+
+    validated, actions = _validate_and_generate_actions_for_repositories(add_repository)
+    if not validated:
+        raise CLIError("Rule {} has invalid syntax.".format(actions))
+    actions.sort()
+
+    if description is not None:
+        description = ' '.join(description)
+
+    resource_group_name = get_resource_group_name_by_registry_name(cmd, registry_name, resource_group_name)
+
+    from msrest.exceptions import ValidationError
+    try:
+        return client.create(
+            resource_group_name,
+            registry_name,
+            scope_map_name,
+            actions,
+            description
+        )
+    except ValidationError as e:
```

Comment:
curious as to why you are catching this exception? Isn't this handled by SDK ? @djyou

Reply:
It prints nicely done. No need for it anymore. Removing it :)